src/integration-test/java/org/openlmis/requisition/web/BatchRequisitionControllerIntegrationTest.java

```diff
@@ -0,0 +1,404 @@
+    checkErrorResponseBody(response, 400);
+  }
+
+  private void checkResponseBody(Response response) throws IOException {
+    String jsonString = response
+        .then()
+        .statusCode(200)
+        .extract()
+        .asString();
+
+    JsonNode json = objectMapper.readTree(jsonString);
+    ArrayNode requisitionErrors = (ArrayNode) json.get("requisitionErrors");
+
+    assertThat(requisitionErrors.size(), equalTo(0));
+
+    ArrayNode requisitionDtos = (ArrayNode) json.get("requisitionDtos");
+
+    List<UUID> retrieved = getIds(requisitionDtos);
+
+    assertThat(retrieved, hasItems(requisitionIds.toArray(new UUID[requisitionIds.size()])));
+  }
+
+  private void checkErrorResponseBody(Response response, int statusCode) throws IOException {
+    String jsonString = response
+        .then()
+        .statusCode(statusCode)
+        .extract()
+        .asString();
+
+    JsonNode json = objectMapper.readTree(jsonString);
+    ArrayNode requisitionErrors = (ArrayNode) json.get("requisitionErrors");
+
+    assertThat(requisitionErrors.size(), equalTo(1));
+    assertThat(
+        requisitionErrors.get(0).get("requisitionId").asText(),
+        equalTo(requisitions.get(0).getId().toString())
+    );
+
+    ArrayNode requisitionDtos = (ArrayNode) json.get("requisitionDtos");
+
+    List<UUID> retrieved = getIds(requisitionDtos);
+
+    assertThat(retrieved, allOf(
+        not(hasItem(requisitions.get(0).getId())),
+        hasItem(requisitions.get(1).getId()),
+        hasItem(requisitions.get(2).getId())
+    ));
+  }
+
+  private List<UUID> getIds(ArrayNode requisitionDtos) {
+    List<UUID> result = Lists.newArrayList();
+    requisitionDtos.forEach(node -> result.add(UUID.fromString(node.get("id").asText())));
+
+    return result;
+  }
+
+  private Response put(String param, List<ApproveRequisitionDto> body) {
+    return startRequest(param)
+        .body(body)
+        .put(RESOURCE_URL);
+  }
+
+  private Response get(String param, List<UUID> ids) {
+    RequestSpecification specification = startRequest(param);
+
+    for (UUID id : ids) {
+      specification = specification.queryParam(ID, id);
+    }
+
+    return specification.get(RESOURCE_URL);
+  }
+
+  private Response post(String param, List<UUID> ids) {
+    RequestSpecification specification = startRequest(param);
+
+    for (UUID id : ids) {
+      specification = specification.queryParam(ID, id);
+    }
+
+    return specification.post(RESOURCE_URL);
+  }
+
+  private RequestSpecification startRequest(String param) {
+    return restAssured
+        .given()
+        .header(HttpHeaders.AUTHORIZATION, getTokenHeader())
+        .contentType(MediaType.APPLICATION_JSON_VALUE)
+        .queryParam(param, "");
+  }
+
+  private void mockRepositorySaveAnswer() {
+    given(requisitionRepository.save(any(Requisition.class))).willAnswer(new SaveAnswer<>());
+  }
+
+  private void mockRequisitionDtoBuilderResponses() {
```

Comment:
I assume it was easier to wrote own builder than mock every external dependency in real one